signinwithapplebutton/src/main/java/com/willowtreeapps/signinwithapplebutton/view/SignInWebViewDialogFragment.kt

```diff
@@ -0,0 +1,124 @@
+import android.webkit.WebView
+import androidx.fragment.app.DialogFragment
+import com.willowtreeapps.signinwithapplebutton.SignInWithAppleCallback
+import com.willowtreeapps.signinwithapplebutton.SignInWithAppleService
+
+@SuppressLint("SetJavaScriptEnabled")
+internal class SignInWebViewDialogFragment : DialogFragment, SignInWithAppleCallback {
+
+    private var authenticationAttempt: SignInWithAppleService.AuthenticationAttempt? = null
+    private var callback: SignInWithAppleCallback? = null
+
+    private val webViewIfCreated: WebView?
+        get() = view as? WebView
+
+    constructor() : super() {
+        authenticationAttempt = null
+    }
+
+    constructor(authenticationAttempt: SignInWithAppleService.AuthenticationAttempt) {
+        this.authenticationAttempt = authenticationAttempt
+    }
+
+    fun configure(
+        callback: SignInWithAppleCallback
+    ) {
+        this.callback = callback
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        if (authenticationAttempt == null) {
+            authenticationAttempt = savedInstanceState?.getParcelable("authenticationAttempt")
+        }
+
+        setStyle(STYLE_NORMAL, android.R.style.Theme_Black_NoTitleBar_Fullscreen)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        super.onCreateView(inflater, container, savedInstanceState)
+
+        val webView = WebView(context).apply {
+            settings.apply {
+                javaScriptEnabled = true
+                javaScriptCanOpenWindowsAutomatically = true
+            }
+        }
+
+        if (authenticationAttempt == null) {
+            Log.e("SIGN_IN_WITH_APPLE", "Authentication attempt is not configured")
+        }
+
+        webView.webViewClient = authenticationAttempt?.let {
+            SignInWebViewClient(it,this@SignInWebViewDialogFragment)
+        }
+
+        if (savedInstanceState != null) {
+            savedInstanceState.getBundle("webView")?.run {
+                webView.restoreState(this)
+            }
+        } else {
+            authenticationAttempt?.run {
+                webView.loadUrl(authenticationUri)
+            }
+        }
+
+        return webView
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+
+        outState.putParcelable("authenticationAttempt", authenticationAttempt)
+
+        outState.putBundle(
+            "webView",
+            Bundle().apply {
+                webViewIfCreated?.saveState(this)
+            }
+        )
+    }
+
+    override fun onStart() {
+        super.onStart()
+
+        dialog?.window?.setLayout(MATCH_PARENT, MATCH_PARENT)
+    }
+
+    // SignInWithAppleCallback
+
+    override fun onSignInWithAppleSuccess(authorizationCode: String) {
+        dialog?.dismiss()
+
+        val callback = callback
+        if (callback == null) {
```

Comment:
think we can simplify this a little bit
`callback?.onSignInWithAppleSuccess(authorizationCode) ?: Log.e("SIGN_IN_WITH_APPLE", "Callback is not configured")`

Reply:
I don't think I'd want to use a log statement's Unit output as a default value. Isn't elvis for expressions, not statements?

Reply:
I'm basing my intuition on how I'd use `??`, the nil-defaulting operator in Swift.